2.1.0/scratch/Dockerfile

```diff
@@ -1,5 +1,5 @@
 FROM scratch
-COPY nats-server /nats-server
+COPY --from=nats:2.1.0-alpine3.10 /nats-server /nats-server
```

Comment:
Help me understand: as of now, we would build the release (with CGO disabled) and save them in this repo. For the scratch image, the original Dockerfile would copy the exe from local repo for the SCRATCH image.
Now you copy from the alpine docker image, which itself is built with downloading the release from the releases GH page, which those are built without CGO disabled (so CGO enabled if you will). Will that not be an issue then?

Reply:
Turns out we already do build fully static binaries. :sweat_smile: 
https://github.com/nats-io/nats-server/blob/master/.goreleaser.yml#L20

With `CGO_ENABLED=0`, we disable CGO. With CGO disabled, we also disable the system DNS resolver. So actually, the `netgo` stuff is superfluous.

```
[nats-server]$ go build
[nats-server]$ ldd nats-server 
    linux-vdso.so.1 (0x00007ffcdd9d9000)
    libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f97b0c6b000)
    libc.so.6 => /usr/lib/libc.so.6 (0x00007f97b0aa4000)
    /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f97b0caf000)

[nats-server]$ CGO_ENABLED=0 go build
[nats-server]$ ldd nats-server 
    not a dynamic executable

[nats-server]$ CGO_ENABLED=0 go build -tags netgo -installsuffix netgo
[nats-server]$ ldd nats-server 
    not a dynamic executable
```

Reply:
Okay... I started to think that I messed-up when adding goreleaser, but checking pre-2.0.4, the scripts/cross_compile.sh was indeed setting CGO_ENABLED=0, so yeah, releases on GH are with CGO disabled..

Reply:
And so if I understand correctly, in docker official images, you will make sure that the alpine is defined first, so that scratch image will copy the exe from there?

Reply:
Yep, correct. Exactly how the Windows images are built.